modular_skyrat/modules/reagent_forging/code/forge.dm

```diff
@@ -80,213 +139,233 @@
-		else
-			icon_state = "forge_full"
+		icon_state = "[primitive ? "primitive_" : ""]forge_full"
 		light_range = 3
 	else if(forge_temperature <= 0)
-		if(primitive)
-			icon_state = "primitive_forge_empty"
-		else
-			icon_state = "forge_empty"
+		icon_state = "[primitive ? "primitive_" : ""]forge_empty"
 		light_range = 0
 
+/**
+ * Here we fix any weird in_use bugs
+ */
 /obj/structure/reagent_forge/proc/check_in_use()
 	if(!in_use)
 		return
 	for(var/mob/living/living_mob in range(1,src))
 		if(!living_mob)
 			in_use = FALSE
 
+/**
+ * Here we spawn coal depending on a chance of using wood
+ */
 /obj/structure/reagent_forge/proc/spawn_coal()
 	new /obj/item/stack/sheet/mineral/coal(get_turf(src))
 
 /obj/structure/reagent_forge/process()
-	if(world_check >= world.time) //to make it not too intensive, every 5 seconds
+	if(!COOLDOWN_FINISHED(src, forging_cooldown)) //every 5 seconds to not be too intensive, also balanced around 5 seconds
 		return
-	world_check += 5 SECONDS
+	COOLDOWN_START(src, forging_cooldown, 5 SECONDS)
 	check_fuel()
 	check_temp()
 	check_in_use() //plenty of weird bugs, this should hopefully fix the in_use bugs
 
+/obj/structure/reagent_forge/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	var/user_smithing_skill = user.mind.get_skill_level(/datum/skill/smithing)
+	switch(user_smithing_skill)
+		if(SKILL_LEVEL_NONE to SKILL_LEVEL_NOVICE)
+			to_chat(user, span_notice("[src] requires you to be more experienced!"))
+		if(SKILL_LEVEL_APPRENTICE to SKILL_LEVEL_JOURNEYMAN)
+			if(forge_level >= FORGE_LEVEL_ONE)
+				to_chat(user, span_warning("Your level of smithing has already influenced [src]!"))
+				return
+			to_chat(user, span_notice("[src] levels up... examine to view what has changed!"))
+			forge_level = FORGE_LEVEL_ONE
+			goliath_ore_improvement = 3
+			playsound(src, 'sound/magic/demon_consume.ogg', 50, TRUE)
+		if(SKILL_LEVEL_EXPERT to SKILL_LEVEL_MASTER)
+			if(forge_level >= FORGE_LEVEL_TWO)
+				to_chat(user, span_warning("Your level of smithing has already influenced [src]!"))
+				return
+			to_chat(user, span_notice("[src] levels up... examine to view what has changed!"))
+			forge_level = FORGE_LEVEL_TWO
+			sinew_lower_chance = 100
+			current_sinew = 10
+			playsound(src, 'sound/magic/demon_consume.ogg', 50, TRUE)
+		if(SKILL_LEVEL_LEGENDARY)
+			if(forge_level >= FORGE_LEVEL_THREE)
+				to_chat(user, span_warning("Your level of smithing has already influenced [src]!"))
+				return
+			to_chat(user, span_notice("[src] levels up... examine to view what has changed!"))
+			forge_level = FORGE_LEVEL_THREE
+			current_core = 3
+			create_reagent_forge()
+			playsound(src, 'sound/magic/demon_consume.ogg', 50, TRUE)
+
 /obj/structure/reagent_forge/attackby(obj/item/I, mob/living/user, params)
 	if(istype(I, /obj/item/stack/sheet/mineral/wood)) //used for weak fuel
+		var/skill_modifier = user.mind.get_skill_modifier(/datum/skill/smithing, SKILL_SPEED_MODIFIER) * 3 SECONDS
```

Comment:
define this variable at the start, before all the if()s, so you can just keep using it throughout rather than re-defining it at the start of each if()